ravenframework/CodeInterfaceClasses/MELCOR/melcorCombinedInterface.py

```diff
@@ -13,104 +13,178 @@
 from ravenframework.CodeInterfaceBaseClass import CodeInterfaceBase
-from .melcorInterface   import MelcorApp
-from .melgenInterface import MelgenApp
+from ..Generic import GenericParser
+import pandas as pd
+
 
 class Melcor(CodeInterfaceBase):
   """
-    this class is used a part of a code dictionary to specialize Model.Code for MELCOR 2. Version
+    This class is used a part of a code dictionary to specialize Model. Code for different MELCOR versions
+    like MELCOR 2.2x, MELCOR 1.86, MELCOR for fusion applications
   """
 
-  def __init__(self):
+  def _readMoreXML(self,xmlNode):
     """
-      Constructor.
-      @ In, None
-      @ Out, None
+      Function to read the portion of the xml input that belongs to this specialized class and initialize
+      some members based on inputs. This can be overloaded in specialize code interface in order to
+      read specific flags.
+      Only one option is possible. You can choose here, if multi-deck mode is activated, from which deck you want to load the results
+      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
+      @ Out, None.
     """
-    CodeInterfaceBase.__init__(self)
-    self.melcorInterface = MelcorApp()
-    self.melgenInterface = MelgenApp()
 
-  def findInps(self,inputFiles):
+    melNode = xmlNode.find('MelcorOutput')
+    varNode = xmlNode.find('variables')
+    plotNode = xmlNode.find('CodePlotFile')
+
+    if varNode is None:
+      raise IOError("Melcor variables not found, define variables to print")
+    if plotNode is None:
+      raise IOError("Please define the name of the MELCOR plot file in the CodePlotFile xml node")
+    if melNode is None:
+      raise IOError("Please enter MELCOR message file name")
+
+    self.varList = [var.strip() for var in varNode.text.split(",")]
+    self.melcorPlotFile = [var.strip() for var in plotNode.text.split(",")][0]
+    self.melcorOutFile = [var.strip() for var in melNode.text.split(",")][0]
+
+    return self.varList, self.melcorPlotFile, self.melcorOutFile
+
+  def findInps(self,currentInputFiles):
     """
       Locates the input files for Melgen, Melcor
-      @ In, inputFiles, list, list of Files objects
-      @ Out, (melgIn,melcIn), tuple, tuple containing Melgin and Melcor input files
+      @ In, currentInputFiles, list, list of Files objects
+      @ Out, (melgIn,melcIn), tuple, tuple containing Melgen and Melcor input files
     """
     foundMelcorInp = False
-    for index, inputFile in enumerate(inputFiles):
+    for index, inputFile in enumerate(currentInputFiles):
       if inputFile.getExt() in self.getInputExtension():
+        if foundMelcorInp:
+          raise IOError(f"Multiple Melcor input files are found {inputFile} and {melgIn}, please check your inputs, only one of input is accepted")
         foundMelcorInp = True
-        melgIn = inputFiles[index]
-        melcIn = inputFiles[index]
+        melgIn = currentInputFiles[index]
+        melcIn = currentInputFiles[index]
```

Comment:
this are the same? why we need to return a tuple if the melgIn and melcIn are the same?